### PR TITLE
Add #TipWithBrave hashtag to twip

### PR DESCRIPTION
**Pull request by emerick**
Fixes brave/brave-browser#4555

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [x] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:

- Enable Brave Rewards
- Visit Twitter web site
- Click the tip icon to tip a particular tweet
- Send the tip
- Press the "Tweet" button to tweet about the tip
- Verify that #TipWithBrave hashtag appears in the posted tweet

Also need to perform the same verification steps when tipping on Twitter from the Rewards panel.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc [hunk 1]
```diff
@@ -216,22 +216,28 @@ std::string MediaTwitter::GetShareURL(
...
   if (!tweet_id->second.empty()) {
     std::string quoted_tweet_url =
         base::StringPrintf("https://twitter.com/%s/status/%s",
                            name->second.c_str(), tweet_id->second.c_str());
-    share_url =
-        base::StringPrintf("https://twitter.com/intent/tweet?url=%s&text=%s",
-                           quoted_tweet_url.c_str(), comment->second.c_str());
+    share_url = base::StringPrintf(
+        "https://twitter.com/intent/tweet?text=%s&url=%s",
+        comment_with_hashtag.c_str(), quoted_tweet_url.c_str());
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc]: nit: why did you switch url and text position?

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc [hunk 2]
```diff
@@ -216,22 +216,28 @@ std::string MediaTwitter::GetShareURL(
   auto comment = args.find("comment");
   auto name = args.find("name");
   auto tweet_id = args.find("tweet_id");
+  auto hashtag = args.find("hashtag");
   if (comment == args.end() || name == args.end() || tweet_id == args.end())
     return std::string();
 
+  // Append hashtag to comment ("%20%23" = percent-escaped space and
+  // number sign)
+  std::string comment_with_hashtag =
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc]: could you please add unit tests for `GetShareURL`

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc (hunk 1 above)
Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc]: I just wanted the order of the replacement parameters in `StringPrintf` to match in the "if" and "else", so it was more obvious what was going on.

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc (hunk 2 above)
Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc]: Yes, done.

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc]: make sense 👍

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.h [hunk 3]
```diff
@@ -30,7 +30,8 @@ class MediaTwitter : public ledger::LedgerCallbackHandler {
   void SaveMediaInfo(const std::map<std::string, std::string>& data,
                      ledger::PublisherInfoCallback callback);
 
-  std::string GetShareURL(const std::map<std::string, std::string>& args);
+  static std::string GetShareURL(
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.h]: Maybe I am missing something, but I don't see change in `bat_get_media` that would reflect this static change

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.h]: Good catch, fixed.

**Comment by emerick** [comment]
@NejcZdovc Fixed linter.

**Pull request merged**